File: android/src/main/kotlin/live/hms/hmssdk_flutter/hls_player/HMSHLSCueExtension.kt
package live.hms.hmssdk_flutter.hls_player

import live.hms.hls_player.HmsHlsCue
import java.text.SimpleDateFormat

class HMSHLSCueExtension {

    companion object{
        fun toDictionary(hmsHlsCue: HmsHlsCue):HashMap<String, String?>{
            val args = HashMap<String, String?>()

            args["id"] = hmsHlsCue.id
            args["start_date"] =  SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(hmsHlsCue.startDate).toString()
            args["end_date"] = SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(hmsHlsCue.endDate).toString()
            args["payload"] = hmsHlsCue.payloadval

            return args
        }
    }
}

File: packages/hmssdk_flutter/ios/Classes/HLSPlayer/HMSHLSCueExtension.swift
//
//  HMSHLSCueExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 24/05/23.
//

import Foundation
import HMSHLSPlayerSDK

class HMSHLSCueExtension {

    static func toDictionary(_ hmsHlsCue: HMSHLSCue) -> [String: Any?] {

        var dict = [String: Any?]()

        dict["id"] = hmsHlsCue.id
        dict["start_date"] = (hmsHlsCue.startDate.timeIntervalSince1970 * 1000)
        if let endDate = hmsHlsCue.endDate {
            dict["end_date"] = (endDate.timeIntervalSince1970 * 1000)
        }
        dict["payload"] = hmsHlsCue.payload

        return dict
    }
}
